sts_synindex_external.R

```diff
@@ -112,83 +118,77 @@ replace_equal_with_underscore <- function(directory_path) {
-# if (nrow(synapse_fileview)>0) {
-#   synapse_manifest_to_upload <- 
-#     synapse_manifest %>% 
-#     dplyr::anti_join(
-#       synapse_fileview %>% 
-#         dplyr::select(parent = parentId,
-#                       s3_file_key = dataFileKey,
-#                       md5_hash = dataFileMD5Hex))
-#   synapse_manifest_to_upload <- 
-#     synapse_manifest_to_upload %>% 
-#     mutate(file_key = gsub("cohort_", "cohort=", file_key),
-#            s3_file_key = gsub("cohort_", "cohort=", s3_file_key))
-# } else {
-#   synapse_manifest_to_upload <- 
-#     synapse_manifest %>% 
-#     mutate(file_key = gsub("cohort_", "cohort=", file_key),
-#            s3_file_key = gsub("cohort_", "cohort=", s3_file_key))
-# }
-# 
-# ## For each file index it in Synapse given a parent synapse folder
-# if(nrow(synapse_manifest_to_upload) > 0){ # there are some files to upload
-#   for(file_number in seq(nrow(synapse_manifest_to_upload))){
-#     
-#     # file and related synapse parent id 
-#     file_= synapse_manifest_to_upload$path[file_number]
-#     parent_id = synapse_manifest_to_upload$parent[file_number]
-#     s3_file_key = synapse_manifest_to_upload$s3_file_key[file_number]
-#     # this would be the location of the file in the S3 bucket, in the local it is at {AWS_PARQUET_DOWNLOAD_LOCATION}/
-#     
-#     absolute_file_path <- tools::file_path_as_absolute(file_) # local absolute path
-#     
-#     temp_syn_obj <- synapser::synCreateExternalS3FileHandle(
-#       bucket_name = PARQUET_BUCKET_EXTERNAL,
-#       s3_file_key = s3_file_key, #
-#       file_path = absolute_file_path,
-#       parent = parent_id
-#     )
-#     
-#     # synapse does not accept ':' (colon) in filenames, so replacing it with '_colon_'
-#     new_fileName <- stringr::str_replace_all(temp_syn_obj$fileName, ':', '_colon_')
-#     
-#     f <- File(dataFileHandleId=temp_syn_obj$id,
-#               parentId=parent_id,
-#               name = new_fileName) ## set the new file name
-#     
-#     f <- synStore(f)
-#     
-#   }
-# }
+# Index files in Synapse --------------------------------------------------
+# Get a list of all files to upload and their synapse locations (parentId)
+STR_LEN_PARQUET_FINAL_LOCATION <- stringr::str_length(PARQUET_FINAL_LOCATION)
+
+## List all local files present (from manifest)
+synapse_manifest <- 
+  read.csv('./current_manifest.tsv', sep = '\t', stringsAsFactors = F) %>%
+  dplyr::filter(!grepl('owner.txt', path)) %>%
+  dplyr::rowwise() %>%
+  dplyr::mutate(file_key = stringr::str_sub(string = path, start = STR_LEN_PARQUET_FINAL_LOCATION)) %>%
+  dplyr::mutate(s3_file_key = paste0(PARQUET_BUCKET_BASE_KEY_ARCHIVE, file_key)) %>%
+  dplyr::mutate(md5_hash = as.character(tools::md5sum(path))) %>%
+  dplyr::ungroup()
+
+# List all files currently indexed in Synapse
+synapse_fileview <- 
+  synapser::synTableQuery(paste0('SELECT * FROM ', SYNAPSE_FILEVIEW_ID))$filepath %>%
```

Comment:
Interesting, what is your rationale to doing this over ?

```
synapser::synTableQuery(paste0('SELECT * FROM ', SYNAPSE_FILEVIEW_ID))$asDataFrame()
```

Reply:
I get the following error when using asDataFrame(). Also, the example for `synapser::synTableQuery()` shows using `...$filepath`.

```
> synapser::synTableQuery(paste0('SELECT * FROM ', SYNAPSE_FILEVIEW_ID))$asDataFrame()

Error in py_call_impl(callable, call_args$unnamed, call_args$named) :  TypeError: 'list' object cannot be interpreted as an integer Run `reticulate::py_last_error()` for details.

Error in py_call_impl(callable, call_args$unnamed, call_args$named) :  RuntimeError: (Unknown exception occurred) Run `reticulate::py_last_error()` for details.
```